File: Assignment37_5.c
// Write a program to reverse the string in place

#include<stdio.h>
#include<stdbool.h>

void Reverse(char *str)
{
    char *start = str;
    char *end = str;
    char temp = '\0';

    while(*end != '\0')
    {
        end++;
    }
    end--;

    while(start < end)
    {
        temp = *start;
        *start = *end;
        *end = temp;
        start++;
        end--;
    }
    end = '\0';

}

int main()
{
    char Arr[] = {'\0'};

    printf("Enter a String: \n");
    scanf("%[^'\n']s",Arr);

    Reverse(Arr);
    
    printf("Updated string is :%s\n",Arr);

    return 0;
}

File: Assignment37_5.java
// Write a program to reverse the string in place

import java.util.*;
class Logic
{
    public void Reverse(String str)
    {
        char Arr[] = str.toCharArray();
        int i = 0;
        

        for(i = Arr.length-1; i >=0; i--)
        {
            System.out.print(Arr[i]); 
                      
        }
    }
}


class Assignment37_5
{
    public static void main(String A[])
    {
        Scanner scanobj = new Scanner(System.in);

        System.out.println("Enter a string: ");
        String sobj = scanobj.nextLine();

        Logic lobj = new Logic();
        lobj.Reverse(sobj);
        
    }
}